chasm/lib/callback/statemachine_test.go

```diff
@@ -0,0 +1,149 @@
+package callback
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.temporal.io/server/chasm"
+	callbackspb "go.temporal.io/server/chasm/lib/callback/gen/callbackpb/v1"
+	"go.temporal.io/server/common/backoff"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/protobuf/proto"
+)
+
+// testMutableContext is a minimal test helper for capturing tasks
+type testMutableContext struct {
+	*chasm.MockMutableContext
+	tasks []testTask
+}
+
+type testTask struct {
+	component  chasm.Component
+	attributes chasm.TaskAttributes
+	message    any
```

Comment:
```suggestion
	payload    any
```

Reply:
Made changes in PR to adjust the references too